.circleci/config.yml

```diff
@@ -0,0 +1,58 @@
+version: 2.0
+jobs:
+    build:
+      docker:
+        - image: madmongo1/fedora-devel:latest
+
+      working_directory: /tmp
+
+      steps:
+        - run:
+            name: fix ssh keys
+            working_directory: ~
+            command: |
+              ssh-keyscan github.com >> githubKey
+              ssh-keygen -lf githubKey
+              cat githubKey >> ~/.ssh/known_hosts
+
+        - run:
+            name: clone boost
+            working_directory: /tmp
+            command: |
+              git clone https://github.com/boostorg/boost.git
+
+        - run:
+            name: update submodules
+            working_directory: /tmp/boost
+            command: |
+              if [ "${CIRCLE_BRANCH}" != "master" ]
+              then
+                git checkout develop
+              fi
+              git submodule update --init --recursive
+
+        - run:
+            name: checkout beast version
+            working_directory: /tmp/boost/libs/beast
+            command: |
+              git remote add circle ${CIRCLE_REPOSITORY_URL}
+              git fetch circle
+              git checkout ${CIRCLE_BRANCH}
+              git submodule update --init --recursive
+
+        - run:
+            name: bootstrap
+            working_directory: /tmp/boost
+            command: ./bootstrap.sh
```

Comment:
You can use this script to grab just the necessary dependencies: https://github.com/boostorg/beast/blob/develop/tools/get-boost.sh

Reply:
I'm wary of that script because each submodule is explicitly named. Submodules seem to come and go in master boost's develop branch.

Reply:
No that's not correct, changes to submodules are infrequent.

Reply:
I was caught out by a change to submodules in boost develop branch just last week!

Reply:
In other words, `int sample_size()` returned 1

Reply:
Thank you for bringing the script to my attention. 

I have used pertinent details from it to improve the throughput of my circleci configuration file.

I have chosen not to use the script as-is for two reasons:

1. It would construct a git tree in the circleci host that does not exactly match the one in my workflow. This would potentially make debugging of issues more difficult for me through circle's ssh debugging interface.

2. The double-entry bookeeping required to keep the script in sync with master and develop of the git root repo is unecessary work. The discovered git repo already has this information intrinsically.

If there are any technical issues that prevent the correct functioning of a circleci build of beast in this PR please kindly bring them to my attention.

If any part of the submitted configu file can be proven subobtimal I would be happy to hear about it.

As things stand I belive this PR is perfectly correct.